File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.java
import java.util.*;

class Solution {
    public int solution(int left, int right) {
        List<Integer> base = new ArrayList<>();
        int sum = 0;

        for (int i = left; i <= right; i++) {
            List<Integer> numList = new ArrayList<>(base);
            for (int j = 1; j <= i; j++) {
                if (i % j == 0) numList.add(j);
            }
            
            if (numList.size() % 2 == 0) sum += i;
            else sum -= i;
        }

        return sum;
    }
}

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.kt
class Solution {
    fun solution(left: Int, right: Int): Int {
        return (left .. right).sumOf {
            var count = 0
            (1..it).forEach { num -> if (it % num == 0) count++ }
            if (count % 2 == 0) +it else -it
        }
    }
}